src/main/scala/io/keen/client/scala/Client.scala

```diff
@@ -381,6 +403,118 @@ trait Writer extends AccessLevel {
     val path = Seq(version, "projects", projectId, "events").mkString("/")
     doRequest(path = path, method = "POST", key = writeKey, body = Some(events))
   }
+
+  /**
+   * Queue events locally for subsequent publishing.
+   *
+   * @param collection The collection to which the event will be added.
+   * @param event The event
+   */
+  def queueEvent(collection: String, event: String): Unit = {
+    val handle: Long = eventStore.store(projectId, collection, event)
+  }
+
+  /**
+   * Schedules sending of queued events if keen.optional.queue.send-interval contains a valid value that
+   * is between MinSendInterval and MaxSendInterval.
+   */
+  private def scheduleSendQueuedEvents(): Unit = {
+
+    sendIntervalSeconds match {
+      // no interval specified, we don't need to schedule events to be sent
+      case i if i == 0 => {
+        // do nothing
+      }
+      // interval is valid, schedule events to be sent
+      case j if j >= MinSendInterval && j <= MaxSendInterval => {
+        // schedule sending from a separate thread at the specific interval
+        val executor = new ScheduledThreadPoolExecutor(1)
```

Comment:
Because I of the scope of this I am pretty sure there is basically no way to shut it down.  I am also concerned it could be GCed almost right away, but am less certain about that.